pageserver/src/layered_repository.rs

```diff
@@ -1486,7 +1486,8 @@ impl LayeredTimeline {
             drop(layers);
             drop(write_guard);
 
-            let mut this_layer_uploads = self.evict_layer(oldest_layer_id)?;
+            let mut this_layer_uploads =
+                self.evict_layer(oldest_layer_id, checkpoint_distance == 0)?;
```

Comment:
I think it might be preferable to pass CheckpointConfig through to `checkpoint_internal` so we can check on `Forced` explicitly. Not absolutely needed tho